sysconfig-formula/_states/suse_sysconfig.py

```diff
@@ -64,12 +64,13 @@ def sysconfig(name, key_values, fillup=None, header_pillar=None, quote=True, quo
       key = key.upper()
     if unbool and isinstance(value, bool):
       value = boolmap[value]
-    if quote and not value.startswith(quote_char) and (
+    is_int = isinstance(value, int)
+    if quote and not ( not is_int and value.startswith(quote_char) ) and (
```

Comment:
Looking at this change, `if quote and (is_int or not value.startswith(quote_char) ) and (` will do the same, but is easier to understand.

Since this condition is non-obvious and only understandable when reading the commit message, a comment saying `integers don't support .startswith()` would be helpful.

Actually I wonder if you could move the `startswith` to the `quote_strings ...` section, which would make things much easier - that would only leave `if quote and (` in the outer condition. (Only strings can start with a quote_char, right?)

Reply:
Hi,

thanks for the input. I tested your last paragraph and added some changes. It needed some additional handling for booleans, but I think it is still better.